python/ray/serve/metric/client.py

```diff
@@ -0,0 +1,156 @@
+            label_names(Optional[Tuple[str]]): The set of label names to be
+                added when recording the metrics.
+
+        Usage:
+        >>> client = MetricClient(...)
+        >>> counter = client.new_counter(
+                "http_counter",
+                description="This is a simple counter for HTTP status",
+                label_names=("route", "status_code"))
+        >>> counter.labels(route="/hi", status_code=200).add()
+        """
+        return self._new_metric(name, MetricType.COUNTER, description,
+                                label_names)
+
+    def new_measure(self,
+                    name,
+                    *,
+                    description: Optional[str] = "",
+                    label_names: Optional[Tuple[str]] = ()):
+        """Create a new measure.
+
+        Measure instruments are independent. They cannot be combined as with
+        counters. Measures can be aggregated after recording to compute
+        statistics about the distribution along selected dimension.
+
+        Args:
+            name(str): The unique name for the measure.
+            description(Optional[str]): The description for the measure.
+            label_names(Optional[Tuple[str]]): The set of label names to be
+                added when recording the metrics.
+
+        Usage:
+        >>> client = MetricClient(...)
+        >>> measure = client.new_measure(
+                "latency_measure",
+                description="This is a simple measure for latency in ms",
+                label_names=("route"))
+        >>> measure.labels(route="/hi").record(42)
+        """
+        return self._new_metric(name, MetricType.MEASURE, description,
+                                label_names)
+
+    def record_event(self, record: MetricRecord):
+        """Called by Counter or Measure to record one observation"""
+        self.metric_records.append(record)
+
+    def _new_metric(
+            self,
+            name,
+            metric_type: MetricType,
+            description: str,
+            label_names: Tuple[str] = (),
+    ):
+        if name in self.registered_metrics:
+            raise ValueError(
+                "Metric with name {} is already registered.".format(name))
+
+        if not isinstance(label_names, tuple):
+            raise ValueError("label_names need to be a tuple, it is {}".format(
+                type(label_names)))
+
+        metric_metadata = MetricMetadata(
+            name=name,
+            type=metric_type,
+            description=description,
+            label_names=label_names,
+            default_labels=self.default_labels.copy(),
+        )
+        metric_class = convert_event_type_to_class(metric_type)
+        metric_object = metric_class(
+            client=self, name=name, label_names=label_names)
+
+        self.registered_metrics[name] = metric_metadata
+        return metric_object
+
+    async def _push_once(self):
+        if not len(self.metric_records):
```

Comment:
nit: I prefer len() == 0 for clarity given that this isn't really any more concise